ipld/unixfs/mod/dagmodifier_test.go

```diff
@@ -867,3 +872,655 @@ func getOffset(reader uio.DagReader) int64 {
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Configure fallback hash (simulating what MFS does)
+		dmod.Prefix.MhType = util.DefaultIpfsHash
+		dmod.Prefix.MhLength = -1
+
+		// Modify first few bytes - this won't change the size but validates
+		// that our check works for any modification
+		modData := []byte("MODIFIED")
+		n, err := dmod.WriteAt(modData, 0)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if n != len(modData) {
+			t.Fatalf("Expected to write %d bytes, wrote %d", len(modData), n)
+		}
+
+		// Sync the changes
+		err = dmod.Sync()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Get result
+		resultNode, err := dmod.GetNode()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// The key verification: if it's a leaf node and would exceed verifcid.DefaultMaxIdentityDigestSize,
+		// it must not use identity hash
+		if len(resultNode.Links()) == 0 {
+			if protoNode, ok := resultNode.(*dag.ProtoNode); ok {
+				encodedData, _ := protoNode.EncodeProtobuf(false)
+				if len(encodedData) > verifcid.DefaultMaxIdentityDigestSize && resultNode.Cid().Prefix().MhType == mh.IDENTITY {
+					t.Errorf("Leaf node with %d bytes must not use identity hash", len(encodedData))
+				}
+			}
+		}
+	})
+
+	t.Run("small identity CID remains identity", func(t *testing.T) {
+		// Create a small UnixFS file node with identity hash
+		// Keep it well under verifcid.DefaultMaxIdentityDigestSize (the identity hash limit)
+		smallData := []byte("hello world")
+
+		// Create a UnixFS file node with identity CID
+		smallNode := makeIdentityNode(smallData)
+
+		// Store the node
+		err := dserv.Add(ctx, smallNode)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Create DagModifier with default splitter
+		dmod, err := NewDagModifier(ctx, smallNode, dserv, testu.SizeSplitterGen(512))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Write small amount of data (total still under verifcid.DefaultMaxIdentityDigestSize)
+		additionalData := []byte(" and more")
+		n, err := dmod.WriteAt(additionalData, int64(len(smallData)))
+		if err != nil {
+			t.Fatalf("WriteAt failed: %v", err)
+		}
+		if n != len(additionalData) {
+			t.Fatalf("expected to write %d bytes, wrote %d", len(additionalData), n)
+		}
+
+		// Get the resulting node
+		resultNode, err := dmod.GetNode()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Verify it still uses identity hash
+		resultCID := resultNode.Cid()
+		if resultCID.Prefix().MhType != mh.IDENTITY {
+			t.Fatalf("expected identity hash for small data, got %v", resultCID.Prefix().MhType)
+		}
+	})
+
+	t.Run("preserves all prefix fields when switching from identity", func(t *testing.T) {
+		// Create initial data that fits well within identity CID limit
+		// This needs to be large enough to trigger modifyDag but small enough
+		// to fit in identity with UnixFS overhead
+		initialData := make([]byte, 100)
+		for i := range initialData {
+			initialData[i] = byte('a' + i%26)
+		}
```

Comment:
See `makeTestData` above.
```suggestion
		initialData := makeTestData(100)
```